### PR TITLE
#1195 Can only activate on mondays

### LINKED ISSUE #1195: Enforce Monday Start Date on Work package activation
### Description

We want the engineers to only be able to have their start dates on mondays, so we need to limit the date picker options to only be mondays when they select a start date in the work package activation modal. 

### Acceptance Criteria

only allows users to select monday

### Proposed Solution

look at how we do it in work package creation and copy and paste the logic into the work package activation modal

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by jackturner106**
## Changes

The work package activation form needed to only allow work package activation on Monday. I copied over the logic from work package creation form to disallow users from selecting any date that is not a Monday. Work package activation had previously stored the date as a string, so I changed the String to a date object to make it easier to work with. 

## Test Cases

I ran the website locally and tested submitting a work package activation, and verified that I was only allowed to select Monday dates. I also verified that changing the Date from a string to a Date object did not mess up form submissions.

## Screenshots

Its not super easy to tell in the screenshots but every date that isn't a monday is greyed out

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #1195

### CODE REVIEW

**Review by tigerhippo — CHANGES_REQUESTED** [review]
This looks great! Awesome job Jack, keep it up!

File: src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx [hunk 1]
```diff
@@ -62,19 +60,17 @@ const ActivateWorkPackageModal: React.FC<ActivateWorkPackageModalProps> = ({
...
-      confirmDetails: false,
-      startDate
-    }
-  });
 
   const defaultValues: FormInput = {
-    startDate: new Date(),
+    startDate: startDate,
     confirmDetails: false
   };
```
Comment (tigerhippo) [inline src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx:68]: You can probably get rid of the defaultValues on line 64 and on line 71, do defaultValues: { confirmDetails: false, startDate: startDate } to simplify the code.

**Review by tigerhippo — COMMENTED** [review]
Looks great, one tiny change and you're good!

File: src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx [hunk 2]
```diff
@@ -85,6 +95,10 @@ const ActivateWorkPackageModal: React.FC<ActivateWorkPackageModalProps> = ({
     reset(defaultValues);
   };
 
+  const disableStartDate = (startDate: Date) => {
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx:98]: rename to isStartDateDisabled, it's a little more precise

**Review by RChandler234 — CHANGES_REQUESTED** [review]
Looking good!

File: src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx [hunk 3]
```diff
@@ -39,25 +40,35 @@ interface ActivateWorkPackageModalProps {
...
+
+  const defaultValues: FormInput = {
+    startDate: startDate,
+    confirmDetails: false
+  };
+
   const { reset, handleSubmit, control } = useForm<FormInput>({
     resolver: yupResolver(schema),
-    defaultValues
+    defaultValues: defaultValues
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx:71]: you can shorten this to just defaultValues, typescript knows to match up the attribute name and the const you've declared in this file

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by RChandler234 — APPROVED** [review]
